p_steps/step_12_2_run_scri.R

```diff
@@ -0,0 +1,252 @@
+# Program Information  ----------------------------------------------------
+
+# Program:      step_12_2_run_scri.R 
+# Author:       Anna Schultze; Ema Alsina, Sophie Bots, Ivonne Martens 
+# Description:  calls a function which runs an SCRI on specified datasets 
+#               runs on all datasets in g_intermediate/scri 
+# Requirements: 
+#               dependencies: preceding steps, packages below 
+#               input:  g_intermediate/scri/*
+#               parameters: in 07_scri_inputs.R 
+#               output: g_output/scri/*model_output.csv
+
+# Housekeeping  -----------------------------------------------------------
+# install and load packages 
+if (!require("pacman")) install.packages("pacman")
+pacman::p_load(packages, character.only = TRUE)
+
+# ensure required folders are created 
+dir.create(file.path("./g_intermediate/scri"), showWarnings = FALSE, recursive = TRUE)
+dir.create(file.path("./g_output/scri"), showWarnings = FALSE, recursive = TRUE)
+
+# get a list of all the datasets 
+
+# Functions ---------------------------------------------------------------
+
+#' FUNCTION 1. run scri
+#' takes a case series csv and fits unadjusted and adjusted scri 
+#' outputs a csv of results for all coefficients 
+#' 
+#' @param brand = vaccine brand, as quoted character
+#' @param outcome = outcome, as quoted character 
+#' @param outcome_name = outcome display name, as quoted character
+#' 
+#' relies on the standardsccs from sccs package, which must be loaded 
+#' note; date cuts specified in the standardsccs function are inclusive of both start and end 
+#' that is, the difference between date1 and date1 is 1 day. 
+#' variables are as follows: 
+#' start = dose 1 - 90 
+#' vd = start pre-exposure window (dose 1 - 30)
+#' vd1 = dose 1 
+#' vd2 = dose 2 
+#' vd3 = start of the "in between doses" period, missing if no dose 2
+#' r1end = min(dose1 + 28, dose2, end)
+#' r2end = min(dose2 + 28, end)
+#' end = min(death, dose1 + 28, dose2 + 28, study_end)
+#' note, have included vd3 as min of d1 risk end and d2, otherwise chops up risk window 2 (which doesn't matter hugely, but for clarity)
+#' note, in the long data, risk periods of interest are numbered 3 and 6, essentially the number of the input vars in the below vector
+
+run_scri <- function(inputdata) {
+  
+  # Import Data -------------------------------------------------------------
+  scri_input <- read.csv(inputdata)
+
```

Comment:
So wouldn't put read.csv(inputdata) in the function but before, and then use the 'inputdata' parameter in your function to refer to the dataset you just read in-- agreed

Reply:
keen to discuss this; how would you approach programming this so that it reads in every single file in a folder, and then calls the macro for each of those files, and then appends the resulting datasets into one? I was trying to avoid manually calling the datasets, and loops.